common/enableRounds/enableRoundsHandler.go

```diff
@@ -0,0 +1,52 @@
+package enableRounds
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/atomic"
+	"github.com/ElrondNetwork/elrond-go/config"
+)
+
+const checkValueOnExecByCallerName = "CheckValueOnExecByCaller"
+
+type enableRoundsHandler struct {
+	*flagsHolder
+}
+
+// NewEnableRoundsHandler creates a new enable rounds handler instance
+func NewEnableRoundsHandler(args config.RoundConfig) (*enableRoundsHandler, error) {
+	checkValueOnExecByCaller, err := getRoundConfig(args, checkValueOnExecByCallerName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &enableRoundsHandler{
+		flagsHolder: &flagsHolder{
+			checkValueOnExecByCaller: checkValueOnExecByCaller,
+		},
+	}, nil
+}
+
+func getRoundConfig(args config.RoundConfig, configName string) (*roundFlag, error) {
+	for _, activationRound := range args.RoundActivations {
```

Comment:
perhaps `RoundActivations` from `RoundConfig` can be a `map[string]ActivationRound` with key `activationRound.Name`.. this way we avoid this for loop

Reply:
:+1:

Reply:
good suggestion, implemented